source/guides/basics/mechanisms.md

```diff
@@ -1,49 +1,150 @@
+
+Items are, fundamentally, their material and their properties. Any item a player has in their inventory can be parsed in this manner, and an item's properties can be read and modified as desired.
 
 #### Material Properties
 
-#### Entities Have Properties Too
+Materials, like items, lack unique identities. A stone block is just a material at a location. A wheat block is only slightly more complicated - it's a material at a location with an `age` property. As you saw above in the `adjustblock` example, using the `age` mechanism allows you to modify that property for a block in the world. When you read block materials from locations, they will return any associated properties. This could be used, for example, to check whether a campfire is sending up smoke signals, or various redstone properties, or even the faces of a mushroom block.
+
+Materials with properties, when parsed by a tag like `<player.cursor_on.material>`, will return an output like `m@wheat[age=7]` for a fully-grown wheat block. In this case, the `age` of the wheat is `7`.
+
+#### Entity Properties
+
+Entities have properties too! Entity properties can be read and modified just like material properties, and you can see a list of all of an entity's properties with the `<EntityTag.describe>` tag. For example, here's a sample output of `/ex narrate <player.describe>` as run by the author of this page:
+
+```
+e@player[has_ai=false;attributes=li@el@GENERIC_MAX_HEALTH/20.0|el@GENERIC_FOLLOW_RANGE/32.0|el@GENERIC_KNOCKBACK_RESISTANCE/0.0|el@GENERIC_MOVEMENT_SPEED/0.10000000149011612|el@GENERIC_ATTACK_DAMAGE/1.0|el@GENERIC_ATTACK_SPEED/4.0|el@GENERIC_ARMOR/0.0|el@GENERIC_ARMOR_TOUGHNESS/0.0|el@GENERIC_LUCK/0.0|;equipment=li@i@air|i@air|i@air|i@air|;health_data=20.0/20.0;inventory_contents=li@i@wooden_axe|i@compass|i@bow|;speed=0]
+```
+
+As you can see above, all of the data about the player is captured by Denizen and output in the format of an EntityTag with various properites. Each property is separated by a semicolon, and the data stored in each property varies based on the nature of the property. 
 
-**TODO: Explain properties more, including the `entity.describe` tag, the idea of tag+mechanism pairs, etc**
+These properties can be adjusted using the `adjust` command or read by using their corresponding tags. For example, `<EntityTag.health>` and `<EntityTag.health_max>` are the tags associated with the `health_data` EntityTag mechanism. `/ex narrate "<player.name> has <player.health>/<player.health_max> health"` will narrate, for example, `Wahrheit has 20/20 health` is an example output based on the data above.
```

Comment:
Health is a bad choice of example here, given the blatant mismatch between the two tags and 1 mechanism.